controller/internal/enforcer/nfqdatapath/datapath_tcp.go

```diff
@@ -326,8 +328,14 @@ func (d *Datapath) processApplicationSynPacket(tcpPacket *packet.Packet, context
 	// Create TCP Option
 	tcpOptions := d.createTCPAuthenticationOption([]byte{})
 
+	// We now generate the claims header
+	compressionType := d.secrets.(*secrets.CompactPKI).Compressed.CompressionTypeToMask()
```

Comment:
It is better to have it optional because for example the tcp proxy doesn't need to have the datapath version, but as we discussed I've added both (version and compression) the fields in the complete datapath handshake